### PR TITLE
Feat/multi condition rules

### LINKED ISSUE #712: Multi-Condition Rules feature
This feature brings the support for multi-condition rules, which literally
means you can pass multiple conditions to trigger a single rule. The change
is backward-compatible and would ideally be shipped by mid-July or end-of-July
and it's gonna be deployed by June 19th for us to implement.

1. The feature is about exposing a new `conditions` array of rule conditions
in the rule object. The existing `condition` field remains for
backward-compatibility.
2. Also, please update the CTS like so: https://github.com/algolia/algoliasearch-client-specs-internal/pull/126/files.
3. Mark the `condition` key as deprecated.

### PR DESCRIPTION
**Pull request by aseure**
Close #712

### CODE REVIEW

File: src/Algolia.Search/Models/Rules/Rule.cs [hunk 1]
```diff
@@ -40,6 +40,11 @@ public class Rule
         /// </summary>
         public Condition Condition { get; set; }
```
Comment (nunomaduro) [inline src/Algolia.Search/Models/Rules/Rule.cs:41]: Can you mark this one as deprecated?

Comment (aseure) [reply src/Algolia.Search/Models/Rules/Rule.cs:42]: Good point, thank you for reminding me. Doing it right now for C# and Go PRs.

**Comment by aseure** [comment]
@Ant-hem 

> Curiosity: Is this `dotnet format` that modified all the comment header?

Yes indeed. Apparently, there was a `U+FEFF` [Zero Width No-Break Space Unicode character](https://www.fileformat.info/info/unicode/char/feff/index.htm) on those lines which was removed by `dotnet format`, as you can see better from my local diff:

Screenshot 2020-07-15 at 14 54 12

**Pull request merged**